apps/server/src/modules/account/services/account.service.ts

```diff
@@ -34,119 +48,267 @@ export class AccountService extends AbstractAccountService {
 		}
 	}
 
-	async findById(id: string): Promise<AccountDto> {
+	public async updateMyAccount(user: User, account: Account, updateData: UpdateMyAccount) {
+		if (account.systemId) {
+			throw new ForbiddenOperationError('External account details can not be changed.');
+		}
+
+		if (!updateData.passwordOld || !(await this.validatePassword(account, updateData.passwordOld))) {
+			throw new AuthorizationError('Your old password is not correct.');
+		}
+
+		let updateUser = false;
+		let updateAccount = false;
+		if (updateData.passwordNew) {
```

Comment:
Define a updateType with all keys optional, add one if it is passed.
Add check to the end if one of the key values !== undefined.
If yes execute a update. To check this you can add a Guard like XXXGuard.checkSomeKeyIsDefined(updateObj);